bgpv1: Add BGP peer password

dgl opened the pull request:
Add 'authSecret' to the BGP peering policies CRD.

This makes it possible to provide passwords for BGP peers through a Secret and CiliumBGPPeeringPolicy.

```release-note
Support BGP passwords in the Go BGP implementation.
```

dylandreimerink reviewed (dismissed):
First of all, thank you very much for this contribution. 

While this is a sorely needed feature, I think this implementation is not secure enough. Having the password in plain text in the policy seems irresponsible. I would suggest updating this PR so the peering policy holds the name+key of a [secret](https://kubernetes.io/docs/concepts/configuration/secret/). Then retrieve this secret from the API server when adding the neighbor.

We already have facilities for getting secrets from the API server which are used in other locations in the code base: https://github.com/cilium/cilium/blob/e8b9edbc4a650d510cd934112ab240deaf4a01e6/pkg/crypto/certificatemanager/certificate_manager.go#L42

asauber commented on pkg/bgpv1/agent/controller.go:
@dgl If we move the changes made in this file to `pkg/bgpv1/gobgp/server.go` (and the initialization into `NewServerWithConfig`, then it seems that we can avoid needing to parameterize the call chain in the reconciler (with the many additional `sf agent.SecretFetcher` parameters), simplifying the implementation quite a bit.

tommyp1ckles commented on pkg/bgpv1/cell.go:
```diff
@@ -54,3 +56,24 @@ func newBGPPeeringPolicyResource(lc hive.Lifecycle, c client.Clientset, dc *opti
+
+	// Do not create this resource if the BGP namespace is not set
+	if dc.BGPSecretsNamespace == "" {
+		return nil
```
Would it make sense to log that this occured

dgl replied:
Yes, sorry, I hadn't considered how this would be configured, but now I've made the helm chart changes it clear we'd expect this to be configured in most cases, so a log does make sense.

tommyp1ckles commented on pkg/bgpv1/gobgp/server.go:
```diff
@@ -107,19 +112,30 @@ func NewServerWithConfig(ctx context.Context, startReq *gobgp.StartBgpRequest) (
+		if err != nil {
+			return fmt.Errorf("failed to fetch secret %q: %w", n.AuthSecret, err)
+		}
+		authPassword = string(secret["passsword"])
```
typo
```suggestion
		authPassword = string(secret["password"])
```

tommyp1ckles commented on pkg/bgpv1/agent/controller.go:
```diff
@@ -176,6 +196,18 @@ func (c *Controller) Start(startCtx hive.HookContext) error {
 
+	secrets, err := c.SecretResource.Store(startCtx)
+	if err != nil {
+		return fmt.Errorf("SecretResource.Store(): %w", err)
```
Could we provide an error message explaining what the issue is rather than just the function name?

dgl replied:
I'm doing the same as the error above it; I think this would only happen in the case of an internal error and I can't really think of a way to explain that, so just copying the error seems reasonable.

tommyp1ckles commented on pkg/bgpv1/agent/controller.go:
```diff
@@ -176,6 +196,18 @@ func (c *Controller) Start(startCtx hive.HookContext) error {
 		return store.List(), nil
 	})
 
+	secrets, err := c.SecretResource.Store(startCtx)
```
If we're only ever getting the secret from the cached store here once, should the underlying watcher continue running?

Would the intention be to continue to watch for updates and reload the secret?

dgl replied:
Yes, but the idea isn't to reload the secret, as if the BGP connection is working it doesn't make sense to tear it down just to update the password. Hence the watcher will update in the background and the secret may be fetched again on a new connection.

kaworu commented on install/kubernetes/cilium/templates/cilium-secrets-namespace.yaml:
I'm worried about the maintenance complexity here.

When [the Envoy config secret namespace was introduced](https://github.com/cilium/cilium/pull/26005), it spawned a discussion about how we could handle namespace generation in a cleaner way. In particular, [@squeed suggestion](https://github.com/cilium/cilium/pull/26005#discussion_r1224206212) to use a dict of desired namespace to be created would result in less code and ease maintenance. Could you please refactor this file accordingly?

dgl replied:
Agreed, I've used the suggestion from the other PR now.

kaworu commented on install/kubernetes/cilium/values.yaml:
```diff
@@ -395,6 +395,12 @@ bgpControlPlane:
+    # -- Create secrets namespace for BGP secrets.
+    create: true
+    name: cilium-bgp-secrets
+
```
[This is a generated file](https://github.com/cilium/cilium/blob/af4d7619ecada7650f489d92c55b9a91d6317a29/install/kubernetes/cilium/values.yaml#L1-L2), please report your change to `install/kubernetes/cilium/values.yaml.tmpl` and then run

```
make -C install/kubernetes cilium/values.yaml
```

dgl replied:
Oops, missed that. Fixed.

kaworu reviewed (changes requested):
Couple of comments, other than that Helm changes LGTM.

kaworu commented on install/kubernetes/cilium/templates/cilium-secrets-namespace.yaml:
thank you, much cleaner now!

kaworu reviewed (approved):
Thanks for the update @dgl, Helm changes LGTM now.

YutaroHayakawa commented on pkg/bgpv1/manager/reconcile.go:
```diff
@@ -299,16 +301,37 @@ func neighborReconciler(
+				return fmt.Errorf("secret %q: missing password key", n.AuthSecret)
+			}
+			l.Debugf("Using TCP password from secret %v", n.AuthSecret)
+		}
```
1. How can we apply changes to the password secret? In this case, the secret is only fetched when we add/update the neighbor settings. Should we watch the changes to the secret file and trigger reconciliation on update instead?
2. Secret fetching logic is duplicated among add and update cases. Can we consolidate them into a single function?

dgl replied:
Originally I was thinking it would be useful to rotate passwords without sessions dropping. I think now there's graceful restart support it's better to just reconnect when the password is changed. It should be a pretty rare occurrence anyway.

I've updated the code to trigger a reconcile on secret changes now.

YutaroHayakawa commented on pkg/k8s/apis/cilium.io/v2alpha1/bgpp_types.go:
```diff
@@ -124,6 +124,10 @@ type CiliumBGPNeighbor struct {
+	// AuthSecret is the name of the secret to use to fetch a TCP authentication
+	// password for this peer.
+	// +kubebuilder:validation:Optional
+	AuthSecret string `json:"authSecret"`
```
Could you follow the k8s API convention and name it `authSecretRef`?

https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#naming-of-the-reference-field

dgl replied:
Done.

YutaroHayakawa replied:
Sorry, one more k8s styling ask. An optional field should be a pointer.

https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#naming-of-the-reference-field

YutaroHayakawa commented on pkg/bgpv1/agent/controller.go:
```diff
@@ -144,6 +156,8 @@ type Controller struct {
 	// and provides a declarative API for configuring BGP peers.
 	BGPMgr BGPRouterManager
 
+	SecretsNamespace string
```
Could you elaborate more about the reason for introducing this (and the Helm option)? I guess it is introduced to reduce the attack surface by limiting the namespace that can be accessed from BGP CPlane, but it's worth noting the reason for the broader users/developers.

dgl replied:
I've added a note in the docs. As you guessed it means the helm chart can safely set up default permissions, without potentially accidentally exposing other secrets in the cluster. It's also easier to just watch a namespace and this shouldn't change too often.

YutaroHayakawa reviewed (changes requested):
Thank you very much for your effort! I went through the changes and added couple of comments. Please take a look 🙏

dgl reviewed (commented):
@YutaroHayakawa Thanks for the comments. I've updated the implementation to reconcile on secret changes now.

YutaroHayakawa commented on pkg/bgpv1/gobgp/server.go:
```diff
@@ -108,6 +108,7 @@ func (g *GoBGPServer) AddNeighbor(ctx context.Context, n types.NeighborRequest)
 	if err != nil {
 		return err
 	}
+
```
Maybe unnecessary.

YutaroHayakawa commented on pkg/bgpv1/manager/reconcile.go:
```diff
@@ -198,7 +200,37 @@ func (r *NeighborReconciler) Priority() int {
+		old := sc.PeerPasswords[key]
+		if old != tcpPassword {
+			l.Debugf("Updated TCP password from secret %q", n.AuthSecretRef)
+			sc.PeerPasswords[key] = tcpPassword
```
I'd prefer separating secret fetch and comparing peer passwords into different functions. Also, changing the server config state inside the function named `fetch` feel a bit odd to me. If we have such a side-effect, then let's change the function name to `update` or something.

YutaroHayakawa reviewed (changes requested):
Did a second look. Thanks for the fix! Added some comments again.

Also, please consider consolidate commits for PR revise into original commits. We prefer clean commit history rather than incremental history of fixes. For example, instead of introducing `authSecret` field in the first commit and fix it to `authSecretRef` in the later commit, please change the commit to introduce `authSecretRef` from the beginning.

asauber commented on pkg/bgpv1/manager/manager.go:
```diff
@@ -105,7 +105,7 @@ func NewBGPRouterManager(params bgpRouterManagerParams) agent.BGPRouterManager {
 // ConfigurePeers should return only once a subsequent invocation is safe.
 // This method is not thread safe and does not intend to be called concurrently.
-func (m *BGPRouterManager) ConfigurePeers(ctx context.Context, policy *v2alpha1api.CiliumBGPPeeringPolicy, cstate *agent.ControlPlaneState) error {
+func (m *BGPRouterManager) ConfigurePeers(ctx context.Context, policy *v2alpha1api.CiliumBGPPeeringPolicy, cstate *agent.ControlPlaneState, secretFetcher types.SecretFetcher) error {
```
Instead of adding `SecretFetcher` as a parameter to this function, we can make it a field on `BGPRouterManager`, and provide it upon initialization of the `BGPRouterManager`. This would significantly simplify the changes in this PR.

asauber replied:
The tests, as included, would be more difficult to implement otherwise. In the future, we should consider a `hive.Cell` for this purpose.

dgl replied:
I spent a while trying to do this, but the issue is the K8s resources are set up in the agent, which already needs a BGPRouterManager before its Start is called. I'm not familiar enough with the hive infrastructure but I can't see a way to break the cycle here.

asauber reviewed (approved):
CLI changes LGTM

nathanjsweet reviewed (approved):
LGTM for API.

dgl committed:
bgpv1: Add BGP peer password support

Add 'authSecretRef' to the BGP peering policies CRD which specifies a
secret which contains a password.

This also adds support for creating the namespace for BGP secrets to the
Helm chart.

Signed-off-by: David Leadbeater <dgl@dgl.cx>

YutaroHayakawa commented on pkg/bgpv1/agent/controller.go:
```diff
@@ -134,6 +145,20 @@ func (c *Controller) Start(startCtx hive.HookContext) error {
+					// Signal the reconciliation logic.
+					c.Sig.Event(struct{}{})
+				}
+				ev.Done(nil)
```
Why don't we handle Delete event?

dgl replied:
The idea was that for manually managed secrets a common workflow is to `kubectl delete` and then `kubectl create`, which would actually bring down a BGP session and then bring it up.

But this touches on your other comment too about handling missing secrets. Having some sort of retry for fetching secrets which doesn't change the state (i.e. eventual reconciliation, much like how Pods work) would solve both. It's a bit hard to do with the current structure as it assumes everything is driven off policy changes, but I'll have a think.

YutaroHayakawa replied:
Sorry, maybe I'm not following. Why can't we just update the secret instead of delete & create? I though if we see the delete event, the right behavior would be do reconcile and shutdown the BGP session. Am I missing something?

dgl replied:
If someone is managing the secret with kubectl directly, then it is a surprisingly complex command to update the secret (see https://stackoverflow.com/questions/45879498/how-can-i-update-a-secret-on-kubernetes-when-it-is-generated-from-a-file) -- I suspect maybe this isn't that common for people using BGP, as BGP is the kind of thing that needs a lot of infrastructure, which to me implies some kind of CD / secrets management system.

But it's also the kind of place where personally if I was dealing with a broken cluster and updating a secret was needed, it would be surprising to me that deleting a secret tears down a BGP session -- secrets or configmaps going away doesn't delete a pod.

I've adjusted the handling now so it handles delete, but if there was previously a valid secret it uses it.

YutaroHayakawa replied:
Ok, I understand. For me, it's ok to go with this behavior. Users still can "delete" a secret by deleting `authSecretRef` and the result is the same as handling the delete event.

YutaroHayakawa commented on pkg/bgpv1/manager/instance.go:
```diff
@@ -38,6 +38,9 @@ type ServerWithConfig struct {
 	// Holds routing policies configured by the policy reconciler.
 	RoutePolicies map[string]*types.RoutePolicy
+
+	// Holds peer passwords, fetched from secrets
```
Nice to have a description of the key string format.

YutaroHayakawa commented on pkg/bgpv1/manager/reconcile.go:
```diff
@@ -295,15 +312,27 @@ func (r *NeighborReconciler) Reconcile(ctx context.Context, p ReconcileParams) e
+			return fmt.Errorf("failed fetching password for neighbor %v %v: %w", n.PeerAddress, n.PeerASN, err)
+		}
+		if r.changedPeerPassword(p.CurrentServer, n, tcpPassword) {
+			r.updatePeerPassword(p.CurrentServer, n, tcpPassword)
```
I think we shouldn't commit this change to `p.CurrentServer` before `UpdateNeighbor` succeeds. Otherwise, when the `UpdateNeighbor` fails, we cannot recover anymore because `changedPeerPassword` is `false` in the next reconciliation.

dgl replied:
Agreed, moved it down.

YutaroHayakawa commented on pkg/bgpv1/manager/reconcile.go:
```diff
@@ -492,6 +521,59 @@ endpointsLoop:
+	if n.AuthSecretRef != "" {
+		secret, err := r.fetchSecret(n.AuthSecretRef)
+		if err != nil {
+			return "", fmt.Errorf("failed to fetch secret %q: %w", n.AuthSecretRef, err)
```
Shouldn't we handle the missing secret case here? The missing secret is an expected state, so we shouldn't treat it as an error here, right?

dgl replied:
Right, I've altered the logic now. It handles missing different to an error.

YutaroHayakawa reviewed (changes requested):
Thanks for the fix and rebasing! I have some fix requests, but I think once we fix this, I don't have any concern anymore.

dgl commented on pkg/bgpv1/gobgp/test.go:
```diff
@@ -0,0 +1,27 @@
+
+// GetExistingPeerForTest is a test only method to fetch the underlying gobgp.Peer.
+func (g *GoBGPServer) GetExistingPeerForTest(peerAddress string, peerASN int64) (*gobgp.Peer, error) {
+	if !testing.Testing() {
```
This is new in go 1.21: https://pkg.go.dev/testing#Testing which means this method can be used for testing but can't be called normally.

For testing this it means it can test the actual password, which seemed easier than fitting into the testing in `pkg/bgpv1/test`, but I'm not hugely attached to this, happy to either keep the API only test (i.e. not test the actual password in unit tests) or potentially something else.

ti-mo replied:
Woah, hold on. Importing `"testing"` has the side effect of declaring all testing CLI flags, which we definitely don't want on the main Cilium agent binary importing this package. This is why `testutils` packages exist. Better would be to implement `GetExistingPeer()` as a free function in a separate testutils subpackage that takes a `GoBGPServer` as an argument.

Looks to me like testing.Testing() was introduced to prevent users from pulling in testutils from non-test code, not the other way around.

dgl replied:
Oh, not quite sure what I was thinking. A `testutils` package doesn't work without making something public which probably shouldn't be. It would be possible to do with build tags, but that makes development annoying (`go test` on its own doesn't work).

I've just reverted this bit, it's possible to infer based on whether the password is set or not if things worked. It would be better to test the rest in an integration test, but it doesn't seem there's one for the BGP control plane (yet?).

ti-mo replied:
> A `testutils` package doesn't work without making something public which probably shouldn't be.

That's fine. In your proposal, GoBGPServer and GetExistingPeerForTest were both exported, so not sure what the difference would be. Which symbols are you worried about specifically? Cilium as a project is not concerned about exporting symbols in general -- the only package with an `internal` dir is hive, all other packages are importable by other modules. You could opt for an `internal/testutils` subpackage if this concerns you.

> It would be possible to do with build tags

After spending weeks removing integration and privileged build tags from tests in favor of envs, that's a hard no from me. :sweat_smile: We want to be able to build all code with a single `go test`. arch-specific files aside.

dgl replied:
> That's fine. In your proposal, GoBGPServer and GetExistingPeerForTest were both exported, so not sure what the difference would be. Which symbols are you worried about specifically? Cilium as a project is not concerned about exporting symbols in general -- the only package with an internal dir is hive, all other packages are importable by other modules. You could opt for an internal/testutils subpackage if this concerns you.

Something in `pkg/bgpv1/gobgp` would have to be exported so something in a testutils can return it, that's why I was trying to add `GetExistingPeerForTest`. Clearly it would be possible to do somehow, but I'm mostly thinking this would be better handled by a (future?) BGP control plane integration test.

YutaroHayakawa replied:
We do have facilities to do tests with the actual peering under [bgpv1/test](https://github.com/cilium/cilium/tree/main/pkg/bgpv1/test). What was missing there?

YutaroHayakawa replied:
@dgl Sorry, I forgot to mention you. Please elaborate on the above so that we can move this forward.

dgl replied:
@YutaroHayakawa I've now made a test that uses bgpv1/test, I had to make diffstore available to bgpv1/agent as well as manager to be able to get the secrets available, hopefully that's okay.

tommyp1ckles reviewed (approved):
Look like this needs another rebase, when I last looked at this the k8s API changes looked ok at a high level however I lack much context on @cilium/sig-bgp matters.

I'm going to approve on behalf of sig-k8s on the assumption that @YutaroHayakawa approves this on behalf of sig-bgp.

ti-mo commented on pkg/option/config.go:
```diff
@@ -2383,6 +2386,9 @@ type DaemonConfig struct {
 	// compatible with MetalLB's configuration.
 	BGPConfigPath string
 
+	// BGPSecretsNamespace
```
I think this comment needs to be fleshed out still.

danehans replied:
+1

I would expect the secret to reside in the same ns as Cilium (`kube-system`) by default. @dgl why does BGP CP auth require a separate ns from the rest of the Cilium components?

dgl replied:
A `CiliumBGPPeeringPolicy` is namespace-less, so it seemed strange to me for something that is namespace-less to refer back to  a hardcoded `kube-system`, users shouldn't need to put anything in kube-system in normal operations.

Also this follows the design used by other components that need a secret like ingresses, envoy, etc. Those are that way because those components create secrets, which the BGP CP currently doesn't do but it seemed sensible to match the pattern.

dgl replied:
Updated the comment.

danehans commented on pkg/k8s/apis/cilium.io/v2alpha1/bgpp_types.go:
```diff
@@ -203,6 +203,10 @@ type CiliumBGPNeighbor struct {
+	// AuthSecret is the name of the secret to use to fetch a TCP authentication
+	// password for this peer.
+	// +kubebuilder:validation:Optional
+	AuthSecretRef *string `json:"authSecretRef"`
```
Consider adding the `omitempty` keyword to omit the field from JSON serialization when unset.

@YutaroHayakawa WDYT about generalizing the auth reference to support other object references in the future? For example:

```go

AuthRef *AuthReference `json:"authSecretRef,omitempty"`

// AuthReference is a reference to a resource that stores auth creds.
type AuthReference struct {
	// Group is the group of the referent. For example, "vendor.io".
	// When unspecified or empty string, k8s core API group is inferred.
	//
	// +optional
	// +kubebuilder:default=""
	Group *string `json:"group"`

	// Kind is the kind of the referent. For example "Secret".
	// When unspecified or empty string, k8s "Secret" kind is inferred.
	//
	// +optional
	// +kubebuilder:default=Secret
	Kind *string `json:"kind"`

	// Namespace is the namespace of the referent.
	// When unspecified or empty string, k8s namespace "kube-system" is inferred.
	//
	// +optional
	// +kubebuilder:default=kube-system
	Namespace *string `json:"namespace"`

	// Name is the name of the referent.
	Name string `json:"name"`
}
```

The above example will default to using a k8s Secret named `name` from the kube-system ns.

dgl replied:
I think hard-coding `kube-system` in the spec like this is problematic, it would actually be the namespace the helm chart is deployed in?

The user may also in that case need a way to configure what namespaces they want RBAC to allow to read from, the nice part about using a single namespace as currently is the permissions are easy to understand and there's no risk of allowing too many permissions.

dgl replied:
I've added `omitempty`.

YutaroHayakawa replied:
IMO, it is a bit overkill for this use case. We just want to pass the password here, and the Secret object is stable and pretty primitive (it's effectively an arbitrary KV). So, I can't come up with any scenario that we need to specify another object. Even if users are managing passwords in a way other than Secret (e.g., SealedSecret or Hashicorp Vault), they usually support generating Secret, so we still don't have to change Group/Kind.

For the namespace, I don't have a strong opinion about how to configure it. In both ways, users need to specify a namespace that the Secret exists and RBAC to allow access.

Currently, it is simple because the namespace is known at installation time, so we can generate the RBAC setting with Helm with a single configuration knob, `secretsNamespace`.

On the other hand, in the way users specify it on CRD, we cannot know the namespace at installation time, so users need to write RBAC manually. However, it gives users flexibility.

Personally, I think we're okay with the current form for the above reason. WDYT?

dgl replied:
I think maybe it would make sense to go for the middle ground of adding `AuthReference`, but *only* putting a `Name` field inside it for now. If later extensions are wanted, then it would be possible to add something like the fields outlined above, without breaking compat.

danehans replied:
+1 on ^ @dgl

danehans replied:
@dgl thanks for adding `omitempty`. Are you still planning to implement https://github.com/cilium/cilium/pull/23759#discussion_r1343613666? Either way, you can consider this conversation resolved since you added `omitempty` and others are fine with the current approach for exposing the auth secret reference.

dgl replied:
I'm going to leave it. when I thought about it actually if it was wanted later we could add a `authRef` field instead, so I don't think there's a need to have an extra struct just to potentially keep compatibility later.

danehans reviewed (commented):
@dgl thank you for working on this feature. When you have a moment, PTAL at my review comments.

YutaroHayakawa reviewed (approved):
Now LGTM 👍 Thanks for your contribution!

rastislavs commented on pkg/bgpv1/agent/controller.go:
```diff
@@ -156,6 +168,20 @@ func (c *Controller) Start(startCtx hive.HookContext) error {
 
+	if c.SecretResource != nil {
+		c.workerpool.Submit("secret-observer", func(ctx context.Context) error {
+			for ev := range c.SecretResource.Events(ctx) {
```
Please note that events for the Secret resource are already signaled by the `diffstore.DiffStore[*slimcorev1.Secret]` see: https://github.com/cilium/cilium/blob/1e41011ee983ceefe19a0c6080be7646faaa74e2/pkg/bgpv1/manager/diffstore.go#L105-L107

Why do we need this event processing here? IMO this code duplicates events for the secrets.

dgl replied:
Oops, I missed it did that, I wrote it before diffstore existed.

rastislavs commented on pkg/bgpv1/agent/controller.go:
```diff
@@ -61,6 +63,12 @@ type Controller struct {
 	// PolicyLister is an interface which allows for the listing of all known policies
 	PolicyLister policyLister
+
+	// SecretResource provides a store of cached policies and allows us to observe changes to the objects in its
```
```suggestion
	// SecretResource provides a store of cached secrets and allows us to observe changes to the objects in its
```

rastislavs commented on pkg/bgpv1/agent/controller.go:
```diff
@@ -85,6 +93,8 @@ type ControllerParams struct {
 	RouteMgr                BGPRouterManager
 	PolicyResource          resource.Resource[*v2alpha1api.CiliumBGPPeeringPolicy]
+	SecretResource          resource.Resource[*slimcorev1.Secret]
+	SecretStore             diffstore.DiffStore[*slimcorev1.Secret]
```
I don't see any use of the `Diff()` method of the `DiffStore` in the change (what is the main reason for using the `DiffStore`). In that case you can use [BGPCPResourceStore](https://github.com/cilium/cilium/blob/1e41011ee983ceefe19a0c6080be7646faaa74e2/pkg/bgpv1/manager/resource_store.go#L18-L22) instead to save some bookkeeping.

rastislavs replied:
Also, why do we need the `SecretStore` here in the `Controller`? I don't see any use of it outside of reconcilers.

dgl replied:
I think that's also a pre-diffstore leftover, it was there because otherwise `.Store()` on Resource would block as it wasn't called from a Start hook. Except I didn't realise it wasn't needed after switching to DiffStore. I should be able to switch to `BGPCPResourceStore` as well, I just hadn't realised that was available.

dgl replied:
Now updated to use BGPCPResourceStore and reverted the diffstore changes that weren't needed, so it's a bit cleaner :)

rastislavs replied:
Thanks a lot, looks much cleaner now! Left one last cleanup nit comment in https://github.com/cilium/cilium/pull/23759/files#r1345298515

rastislavs commented on pkg/bgpv1/agent/controller.go:
```diff
@@ -61,6 +62,10 @@ type Controller struct {
 	PolicyLister policyLister
+	// SecretResource provides a store of cached secrets and allows us to observe changes to the objects in its
+	// store.
+	SecretResource resource.Resource[*slimcorev1.Secret]
```
`SecretResource` became unnecessary in the `Controller` as well, can be removed.

dgl replied:
Removed.

danehans commented on pkg/bgpv1/types/bgp.go:
```diff
@@ -44,6 +44,8 @@ type Path struct {
 type NeighborRequest struct {
 	Neighbor *v2alpha1api.CiliumBGPNeighbor
 	VR       *v2alpha1api.CiliumBGPVirtualRouter
+	// TCPPassword is the "AuthSecret" in the Neighbor, fetched from a secret
```
nit: Are there any other supported password types? If not, consider omitting `TCP` throughout to simplify the naming.

dgl replied:
I've dropped it, it's easy enough to change it later if needed (the other types would be other than "MD5", which we could add a key in the secret for, rather than other auth methods, the password is a bit of a misnomer anyway -- it's actually a passphrase / cipher key, but that's what it is called commonly and in the gobgp API).

rastislavs reviewed (approved):
LGTM 👍 Thanks!
